### PR TITLE
Feat nxp 29932 stream introspection

### PR DESCRIPTION
**Pull request by bdelbosc**

### CODE REVIEW

File: modules/runtime/nuxeo-runtime-stream/src/main/java/org/nuxeo/runtime/stream/StreamMetricsReporter.java [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+        }
+    }
+
+    protected String getNodeId() {
+        if (nodeId == null) {
+            ClusterService clusterService = Framework.getService(ClusterService.class);
+            if (clusterService.isEnabled()) {
+                // register cache invalidator
+                nodeId = clusterService.getNodeId();
+            }
```
Comment (kevinleturc) [inline modules/runtime/nuxeo-runtime-stream/src/main/java/org/nuxeo/runtime/stream/StreamMetricsReporter.java:86]: Is the comment still relevant?

**Review by kevinleturc — COMMENTED** [review]
Grat work 💪

**Review by efge — CHANGES_REQUESTED** [review]

File: modules/core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/introspection/StreamIntrospectionComputation.java [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+        ArrayNode metricsNode = OBJECT_MAPPER.createArrayNode();
+        metricsNode.addAll(metrics.values());
+        node.set("metrics", metricsNode);
+        try {
+            model = OBJECT_MAPPER.writer().writeValueAsString(node);
+        } catch (JsonProcessingException e) {
+            log.error("Cannot build JSON model", e);
+            model = "{}";
+        }
+    }
```
[the remark this answers is not in the document]
Comment (bdelbosc) [reply modules/core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/introspection/StreamIntrospectionComputation.java:182]: yes probably but this is only the early stage of the representation that can be refactored depending on how we want to exploit it, the important thing is to start publishing the information (metrics and processor producer code)

**Comment by bdelbosc** [comment]
PR screwed up, review taken in account in https://github.com/nuxeo/nuxeo/pull/4534

**Pull request closed without merging**